pkg/process/args.go

```diff
@@ -40,12 +46,14 @@ func (t *TargetArgs) Validate() error {
 // ParseTargetArgs returns TargetArgs for the target pointed to by the
 // environment variable OTEL_GO_AUTO_TARGET_EXE.
 func ParseTargetArgs() *TargetArgs {
-	result := &TargetArgs{}
-
-	val, exists := os.LookupEnv(ExePathEnvVar)
-	if exists {
+	if val, exists := os.LookupEnv(ExePathEnvVar); exists {
+		result := &TargetArgs{}
 		result.ExePath = val
+		serviceName, exists := os.LookupEnv(otelServiceNameEnvVar)
+		if exists {
+			result.ServiceName = serviceName
+		}
+		return result
 	}
-
-	return result
+	return nil
```

Comment:
you return `nil` here if the exePath isn't set, which is invalid config, but then you can't call `Validate()` to catch it

maybe change this like:

```go
    result := &TargetArgs{}
	if val, exists := os.LookupEnv(ExePathEnvVar); exists {
		result.ExePath = val
	}
    if serviceName, exists := os.LookupEnv(otelServiceNameEnvVar); exists {
		result.ServiceName = serviceName
	}
	return result
```

breaking the `serviceName` check out of the nested exePath check is a little easier to read too. you could also fail earlier if exePath isn't set, but we have Validate() functions for that anyway

Reply:
what about an explicit field in `targetArgs` that says you want to monitor all services (eg `targetArgs.allTargets = true`) instead of relying on `nil` to imply that's what you're doing?

Reply:
That is good idea. Will add it .